sdk/src/main/scala/io/horizen/account/state/ForgerStakeV2MsgProcessor.scala

```diff
@@ -81,21 +87,75 @@ case class ForgerStakeV2MsgProcessor(networkParams: NetworkParams) extends Nativ
     Array.emptyByteArray
   }
 
-}
+  def doActivateCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
+
+    //Check is well formed
+    requireIsNotPayable(invocation)
+    checkInputDoesntContainParams(invocation)
+
+    //Check it cannot called twice
+    if (StakeStorage.isActive(view)) {
+      val msgStr = s"Forger stake V2 already activated"
+      log.debug(msgStr)
+      throw new ExecutionRevertedException(msgStr)
+    }
+
+    val intrinsicGas = invocation.gasPool.getUsedGas
+
+    //Get all existing stakes from old native contract
+    val result = context.execute(invocation.call(FORGER_STAKE_SMART_CONTRACT_ADDRESS, BigInteger.ZERO,
+      BytesUtils.fromHexString(ForgerStakeMsgProcessor.GetListOfForgersCmd), invocation.gasPool.getGas))
+    val listOfExistingStakes = AccountForgingStakeInfoListDecoder.decode(result).listOfStakes
+    val stakesByForger = listOfExistingStakes.groupBy(_.forgerStakeData.forgerPublicKeys)
+
+    val epochNumber = context.blockContext.consensusEpochNumber
+
+    stakesByForger.foreach { case (forgerKeys, stakesByForger) =>
+      // Sum the stakes by delegator
+      val stakesByDelegator = stakesByForger.groupBy(_.forgerStakeData.ownerPublicKey)
+      val listOfTotalStakesByDelegator = stakesByDelegator.mapValues(_.foldLeft(BigInteger.ZERO){
+        (sum, stake) => sum.add(stake.forgerStakeData.stakedAmount)})
+      //Take first delegator for registering the forger
+      val (firstDelegator, firstDelegatorStakeAmount) = listOfTotalStakesByDelegator.head
+      StakeStorage.addForger(view, forgerKeys.blockSignPublicKey,
```

Comment:
let's remember that in this case we don't have to check the 10ZEN minimum thershold for adding a new forger

Reply:
Added a comment as reminder.